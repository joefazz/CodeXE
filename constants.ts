import { Languages } from './@types';

export const colors = {
    mainBlue: 'dodgerblue',
    mainGreen: '#3ca56e',
    backgroundDark: '#282c34',
    backgroundBlue: '#2d3e5d',
    backgroundDarkTranslucent: 'rgba(40,44,52, 0.7)'
};

export const fonts = {
    display: "'Arvo', serif",
    body: "'Josefin Sans', sans-serif"
};

export const languageOptions: { value: string; label: string }[] = [
    { value: Languages.JS, label: 'JavaScript' },
    { value: Languages.PYTHON, label: 'Python' },
    { value: Languages.C, label: 'C/C++' }
];

export const DOMAIN = `https://midgard.codexe.run`;
