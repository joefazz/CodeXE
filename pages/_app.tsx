import App, { Container, NextAppContext } from 'next/app';
import Head from 'next/head';
import React from 'react';
import { MessageTypes } from '../@types';
import { handleMessage } from '../functions/websockets';

export const SocketContext = React.createContext({});

export default class MyApp extends App {
    containerInfo: any;
    socket: WebSocket;

    static async getInitialProps({ Component, ctx }: NextAppContext) {
        let pageProps = {};

        if (Component.getInitialProps) {
            pageProps = await Component.getInitialProps(ctx);
        }

        return { pageProps };
    }

    state = {
        status: 'disconnected',
        containerName: '',
        id: '',
        activityId: '',
        response: {
            readData: {},
            metaData: { saveInfo: {} },
            writeData: { output: 'Output' }
        }
    };

    componentDidMount() {
        this.socket = new WebSocket('ws://localhost:4000/');

        this.socket.onopen = () => {
            console.log('Socket Opened');
        };

        this.socket.onmessage = (event) => this.setState(handleMessage(event, this.state));

        this.socket.onclose = function() {
            // socket.send('container.end');
            console.log('WebSocket has been closed');
        };
        this.socket.onerror = function(event) {
            console.log('Error', event);
        };
    }

    // TODO: GET THIS WORKING FFS
    componentWillUnmount() {
        this.socket.send(
            JSON.stringify({ message: 'Container.Stop', data: { id: this.state.id } })
        );
        this.socket.close();
    }

    render() {
        const { Component, pageProps } = this.props;
        return (
            <Container>
                <Head>
                    <title>CodeXE</title>
                    <link
                        href="https://fonts.googleapis.com/css?family=Cantarell|Arvo"
                        rel="stylesheet"
                    />
                    <link
                        rel="shortcut icon"
                        href="/static/images/favicon.ico"
                        type="image/x-icon"
                    />
                    <link rel="icon" href="/static/images/favicon.ico" type="image/x-icon" />
                    <link
                        rel="stylesheet"
                        href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.0/normalize.min.css"
                    />
                </Head>
                <SocketContext.Provider value={{ ...this.state, socket: this.socket }}>
                    <Component {...pageProps} />
                </SocketContext.Provider>
            </Container>
        );
    }
}
