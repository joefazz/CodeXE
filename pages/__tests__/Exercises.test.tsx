/* eslint-env: "jest" */

import React from 'react';

import { render } from 'react-testing-library';
import { SocketContext } from '../_app';
import { CONTEXT_DEFAULT } from '../../constants';
import Exercises from '../exercises';
import Exercise from '../exercise';

const EXERCISE_MOCK: Exercise = {
    _id: '0',
    title: 'Python Strings 101',
    description: 'Learn the basics of strings in Python!',
    container: 'python_basics',
    entrypoint: 'main.py',
    length: 3,
    difficulty: 'beginner',
    language: 'python'
};

describe('Exercises Page Snapshot', () => {
    it('Should match Snapshot', () => {
        const { asFragment } = render(
            <SocketContext.Provider value={CONTEXT_DEFAULT}>
                <SocketContext.Consumer>
                    {() => <Exercises exercises={[EXERCISE_MOCK]} />}
                </SocketContext.Consumer>
            </SocketContext.Provider>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Create an exercise', () => {});
