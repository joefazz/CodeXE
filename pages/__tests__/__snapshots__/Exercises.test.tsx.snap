// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercises Page Snapshot Should match Snapshot 1`] = `
<DocumentFragment>
  <div
    class="Layout__RootPage-sc-19ikuwz-0 jmOFgE"
  >
    <header
      class="Header-sc-2vudz7-0 iCnnCf"
    >
      <div
        class="Layout__LeftWrapper-sc-19ikuwz-1 jJmoFf"
      >
        <a
          class="Logo-sc-108q7dg-0 hhrpgI"
        >
          CodeXE
        </a>
        <div
          class="Layout__StatusWrapper-sc-19ikuwz-2 bjSJPB"
        >
          <span
            style="padding-bottom: 2px;"
          >
            Connection Status:
          </span>
           
          <div
            class="Layout__ContainerStatus-sc-19ikuwz-3 idJFII"
            color="red"
          />
        </div>
      </div>
      <div
        style="display: flex; justify-content: space-evenly; align-items: center; width: 40%;"
      >
        <a
          class="NavLink-sc-1yp13yj-0 fHJUjJ"
        >
          Sandbox
        </a>
        <a
          class="NavLink-sc-1yp13yj-0 fHJUjJ"
        >
          Exercises
        </a>
      </div>
    </header>
    <div
      style="background-color: rgb(45, 62, 93); padding-top: 75px; width: 100vw;"
    >
      <div
        class="exercises__Page-sc-71g1vq-0 ivuIJT"
      >
        <section
          class="exercises__List-sc-71g1vq-1 fMgiSC"
        >
          <div
            class="exercises__ExerciseCard-sc-71g1vq-7 kqsRiY"
          >
            <div>
              <section
                class="exercises__WindowButtonWrapper-sc-71g1vq-8 kFqMgY"
              >
                <i
                  class="exercises__WindowButton-sc-71g1vq-9 irkQpI"
                  color="#fc5753"
                />
                <i
                  class="exercises__WindowButton-sc-71g1vq-9 icLVjQ"
                  color="#fdbc40"
                />
                <i
                  class="exercises__WindowButton-sc-71g1vq-9 bftZuI"
                  color="#34c748"
                />
              </section>
              <span>
                Python Strings
              </span>
            </div>
            <code
              class="exercises__ExerciseDescription-sc-71g1vq-10 fAslEs"
            >
              <ul
                style="list-style: none;"
              >
                <li>
                  &gt; Python?!
                </li>
                <li>
                  &gt; Is there a snake?
                </li>
                <li>
                  &gt; I hate snakes!
                </li>
              </ul>
            </code>
          </div>
          <div
            class="exercises__ExerciseCard-sc-71g1vq-7 kqsRiY"
          >
            <div>
              <section
                class="exercises__WindowButtonWrapper-sc-71g1vq-8 kFqMgY"
              >
                <i
                  class="exercises__WindowButton-sc-71g1vq-9 irkQpI"
                  color="#fc5753"
                />
                <i
                  class="exercises__WindowButton-sc-71g1vq-9 icLVjQ"
                  color="#fdbc40"
                />
                <i
                  class="exercises__WindowButton-sc-71g1vq-9 bftZuI"
                  color="#34c748"
                />
              </section>
              <span>
                C++ Pointers
              </span>
            </div>
            <code
              class="exercises__ExerciseDescription-sc-71g1vq-10 fAslEs"
            >
              <ul
                style="list-style: none;"
              >
                <li>
                  &gt; What is a Pointer?
                </li>
                <li>
                  &gt; Sorry what?
                </li>
                <li>
                  &gt; Still don't get it
                </li>
              </ul>
            </code>
          </div>
          <div
            class="exercises__ExerciseCard-sc-71g1vq-7 kqsRiY"
          >
            <div>
              <section
                class="exercises__WindowButtonWrapper-sc-71g1vq-8 kFqMgY"
              >
                <i
                  class="exercises__WindowButton-sc-71g1vq-9 irkQpI"
                  color="#fc5753"
                />
                <i
                  class="exercises__WindowButton-sc-71g1vq-9 icLVjQ"
                  color="#fdbc40"
                />
                <i
                  class="exercises__WindowButton-sc-71g1vq-9 bftZuI"
                  color="#34c748"
                />
              </section>
              <span>
                NodeJS in Anger
              </span>
            </div>
            <code
              class="exercises__ExerciseDescription-sc-71g1vq-10 fAslEs"
            >
              <ul
                style="list-style: none;"
              >
                <li>
                  &gt; What is Node?
                </li>
                <li>
                  &gt; Building a Server
                </li>
                <li>
                  &gt; Advanced Concepts
                </li>
              </ul>
            </code>
          </div>
        </section>
        <section
          class="exercises__CreateArea-sc-71g1vq-2 hCDKJA"
        >
          <h1>
            Create your own exercise!
          </h1>
          <form>
            <label
              for="name"
              id="namelabel"
            >
              Exercise Name
            </label>
            <input
              id="name"
              name="name"
              placeholder="Enter the title of your exercise here"
              type="text"
              value=""
            />
            <label
              for="select"
              id="selectlabel"
            >
              Select Language
            </label>
            <select
              class="Selector-sc-1nm0eq5-0 hEHzPj"
              id="langselect"
            >
              <option
                value="javascript"
              >
                JavaScript
              </option>
              <option
                value="python"
              >
                Python
              </option>
              <option
                value="cpp"
              >
                C/C++
              </option>
            </select>
            <button
              class="Button-sc-1w81v9u-0 kpoddQ"
              type="submit"
            >
              Create Exercise
            </button>
            <label
              for="activities"
              id="activitylabel"
            >
              Activities:
            </label>
            <div
              class="exercises__CreateActivity-sc-71g1vq-3 fBZhSn"
              id="activities"
            >
              <div
                class="exercises__CreateActivityFormWrapper-sc-71g1vq-4 dCPebn"
              >
                <label
                  for="activityname"
                >
                  Activity Name
                </label>
                <input
                  id="activityname"
                  name="activityname"
                  type="text"
                  value="Example"
                />
                <label
                  for="activitydescription"
                >
                  Activity Description
                </label>
                <textarea
                  id="activitydescription"
                  name="activitydescription"
                >
                  Describe your activity here and press the plus to add more :)
                </textarea>
                <label
                  for="activitytask"
                >
                  Task
                </label>
                <input
                  id="activitytask"
                  name="activitytask"
                  type="text"
                  value="Print: Hello world!"
                />
                <label
                  for="prebaked"
                >
                  Placeholder code
                </label>
              </div>
              <div
                class="exercises__NumberWrapper-sc-71g1vq-5 bOwtrs"
              >
                <div
                  style="padding-left: 15%; width: 100%; height: auto; display: flex; flex-direction: row; flex-wrap: wrap;"
                >
                  <div
                    class="exercises__ExerciseNumber-sc-71g1vq-6 iWmlCk"
                  >
                    1
                  </div>
                  <div
                    class="exercises__ExerciseNumber-sc-71g1vq-6 kDuaqT"
                  >
                    +
                  </div>
                </div>
              </div>
            </div>
          </form>
        </section>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
